library/src/lib/calendar/calendar2/calendar2.component.ts

```diff
@@ -297,4 +313,24 @@ export class Calendar2Component implements OnInit, ControlValueAccessor {
         }
     }
 
+    // --------------- CODE CONCERNING THE MONTH VIEW --------------- 
+    /** @hidden */
+    private setupLocalization(): void {
+        this.monthNames = this.calendarI18n.getAllShortMonthNames();
+
+        this.i18nLocalSub = this.calendarI18n.i18nChange.subscribe(() => {
+            this.monthNames = this.calendarI18n.getAllShortMonthNames();
+            this.cd.detectChanges();
+        });
+    }
+
+    /** @hidden */
+    ngOnDestroy() {
+        if (this.i18nLocalSub) {
+            this.i18nLocalSub.unsubscribe();
```

Comment:
I prefer you use the methodology of the takeUntil, it just scales better for large components

Reply:
Me, too. But that's how you implemented it for the calendar and I wanted to avoid discussions :)

Reply:
yes so let's change it since we're modernizing :D